crates/server/README.md

```diff
@@ -0,0 +1,214 @@
+
+**Authentication**: The Admin API facilitates user authentication via selected wallets, currently supporting MetaMask and NEAR networks. Authentication details will be explained in later sections.
+
+**Integration with Web Applications**: The authentication mechanism is also used by web applications designed for applications (loaded wasm) in participating context, ensuring secure and authenticated access.
+
+### 2. JSON rpc
+
+The JSON-RPC component of the Node Server facilitates communication between the clients and the application binaries (contexts) running on node. This allows seamless interaction and data management for applications.
+
+The JSON-RPC interface provides two primary methods:
+
+- Query
+- Mutate
+
+#### Query Method
+The `Query` method retrieves data from the application in participating context. For instance, in the Only Peers forum application, posts and comments stored in the application's storage can be queried using the JSON-RPC interface. This enables users to fetch and display content from the forum.
+
+#### Mutate Method
+The `Mutate` method allows modification of the application's data in participating context. For example, in the Only Peers forum application, users can create new posts or comments. The Mutate method updates the application's storage with these new entries, facilitating dynamic content creation and interaction within the application.
+
+### 3. Websocket
+
+The WebSocket is used for subscribing to and unsubscribing from certain context running in the Node Server. Defined handlers manage subscription states for WebSocket connections, allowing clients to receive updates about specific contexts they are interested in.
+WebSocket handlers are essential for managing real-time subscriptions within the Node Server. They allow clients to dynamically subscribe to and unsubscribe from updates about various application contexts.
+
+#### Subscription Handling:
+
+Websocket handles requests to subscribe to specific contexts and send responses back to the client with the subscribed context IDs.
+
+#### Unsubscription Handling:
+
+Websocket handle requests to unsubscribe from specific contexts and send responses back to the client with the unsubscribed context IDs.
+
+## Admin API endpoints
+
+The Admin API endpoints are split into protected and unprotected routes, where protected routes require authentication.
+
+**Base path**: `/admin-api`
+
+### Protected Routes
+
+These routes require authentication using auth headers.
+Auth headers are generated using `createAuthHeader` function from the `calimero sdk` library.
+
+Parts of the Auth Headers
+ 1. `wallet_type`: Specifies the type of wallet used (e.g., NEAR).
+ 2. `signing_key`: Encoded public key used for signing the request.
+ 3. `signature`: Encoded signature generated from the payload hash.
+ 4. `challenge`: Encoded hash of the payload, serving as a challenge.
+ 5. `context_id`: Optional context identifier for additional request context.
```

Comment:
It is optional only for admin dashboard right? For all other apps it is mandatory